expression/builtin_arithmetic.go

```diff
@@ -762,11 +762,24 @@ func (s *builtinArithmeticIntDivideDecimalSig) evalInt(row chunk.Row) (ret int64
 		return 0, true, err
 	}
 
+	isLHSUnsigned := mysql.HasUnsignedFlag(s.args[0].GetType().Flag)
+	isRHSUnsigned := mysql.HasUnsignedFlag(s.args[1].GetType().Flag)
+
 	ret, err = c.ToInt()
-	// err returned by ToInt may be ErrTruncated or ErrOverflow, only handle ErrOverflow, ignore ErrTruncated.
-	if err == types.ErrOverflow {
-		return 0, true, types.ErrOverflow.GenWithStackByArgs("BIGINT", fmt.Sprintf("(%s DIV %s)", s.args[0].String(), s.args[1].String()))
+	if isLHSUnsigned || isRHSUnsigned {
+		val, err := c.ToUint()
+		// err returned by ToUInt may be ErrTruncated or ErrOverflow, only handle ErrOverflow, ignore ErrTruncated.
+		if err == types.ErrOverflow && ret != 0 {
```

Comment:
After moving `ret, err = c.ToInt()` to line 777, it seems that we don't need to do the check `ret != 0`?